internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -671,15 +671,16 @@ func convertNetworkConfig(network manifest.NetworkConfig) template.NetworkOpts {
 	if placement.IsEmpty() {
 		return opts
 	}
-	if placement.PlacementString == nil {
-		opts.AssignPublicIP = template.DisablePublicIP
-		opts.SubnetIDs = placement.PlacementArgs.Subnets
+	if placement.PlacementString != nil {
+		if *placement.PlacementString == manifest.PrivateSubnetPlacement {
+			opts.AssignPublicIP = template.DisablePublicIP
+		}
+		opts.SubnetsType = subnetPlacementForTemplate[*placement.PlacementString]
 		return opts
 	}
-	if *placement.PlacementString == manifest.PrivateSubnetPlacement {
-		opts.AssignPublicIP = template.DisablePublicIP
-	}
-	opts.SubnetsType = subnetPlacementForTemplate[*placement.PlacementString]
+	opts.AssignPublicIP = template.DisablePublicIP
+	opts.SubnetsType = ""
```

Comment:
tiny nit: is this line here just to be sure we aren't setting `SubnetsType`? we could probably remove it

Reply:
this is actually to unset the subnet types since it is contradictory to subnetIDs.

Reply:
oh lol didn't realize it was set to `template.PublicSubnetsPlacement` by default. I feel like this function probably also needs some refactoring in the future 😂